### DISCUSSION TITLE
WPF中的TextBox输入校验的最佳实现，Validation还是自定义依赖属性？

### DISCUSSION THREAD

**Discussion by switchxiaomu** [BYJRK/DotNet-Discussions#48] [WPF 开发]
大佬，能否讲讲关于WPF中的TextBox控件如何做好输入校验和错误提示？有没有最佳实践？

**Comment by BYJRK** [comment]
推荐你几个视频：

- https://www.youtube.com/watch?v=5KF0GGObuAQ
- https://www.youtube.com/watch?v=7QkeXHRfpj8

大致思路通常有这么几种：
- 最简单的，在 ViewModel 的 setter 里面检查 value 的值是否合适，不合适直接报错，然后前台 TextBox 的 Text 在写 Binding 时写上 `ValidationOnExceptions=True`
- 写 ValidationRule，并想办法在前台提示错误及错误原因
- 写自定义控件、附加属性、行为等

个人认为并没有什么所谓的最佳实践，看情况选择合适且不违背开发模式的方式即可

**Reply by switchxiaomu** [reply]
非常及时且有建设性的答复，谢谢！
